gcs/innards/state.cc

```diff
@@ -1232,6 +1236,18 @@ auto State::infer_on_objective_variable_before_propagation() -> Inference
         return Inference::NoChange;
 }
 
+auto innards::State::add_constraint_state(const ConstraintState c) -> unsigned long
+{
+    // Copying because I'm not sure if making it a reference is a bad idea... (want it to persist)
```

Comment:
Yes, copying is correct.

Reply:
Maybe one day in the future we can switch this to use COWs, but I don't think that's a priority.

Reply:
What are COWs? Aside from the things that eat grass and make milk (just googled it and felt silly)

Reply:
Copy On Write. We do it in `State` to avoid copying a `std::set` on every branch if it's not actually going to be changed.